server/src/session.ts

```diff
@@ -743,6 +743,11 @@ export class Session {
         },
         codeActionProvider: this.ivy ? {
           resolveProvider: true,
+          // Now the Angular code action provider only supports `QuickFix`. If leave the
+          // `codeActionKinds` empty, all action requests will be sent to the Angular language
+          // service, especially for the action before saving the file, the Angular code action
+          // provider will try to fix all errors in the whole file, it's expensive.
```

Comment:
nit: could you also reference the [issue](https://github.com/angular/vscode-ng-language-service/issues/1828) in the comment here so it can be easily looked up